src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -482,44 +482,78 @@ public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
             }
             else
             {
-                if (reader.BaseStream.Length < reader.BaseStream.Position + extraFieldLength + entry.CompressedLength + 4)
+                // There are 4 distinct scenario we would like to support
```

Comment:
This comment says that there are 4 distinct scenarios and then the list has just two top-level items. It is confusing

Reply:
Will update them